include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -49,11 +49,16 @@
 #include <range/v3/view/repeat_n.hpp>
 #include <range/v3/view/slice.hpp>
 
+#include <seqan3/core/concept/cereal.hpp>
 #include <seqan3/core/concept/iterator.hpp>
 #include <seqan3/range/container/concept.hpp>
 #include <seqan3/range/detail/random_access_iterator.hpp>
 #include <seqan3/range/metafunction.hpp>
 
+#if SEQAN_WITH_CEREAL
+#include <cereal/types/vector.hpp>
+#endif
```

Comment:
> I don't see that you use this include.

You need it, when you serialise concatenated_sequences, because it contains a vector as member (likely even two) which are serialised.

> Isn't that included in cereal/cereal.hpp?

No. Support for serialising STL types needs to be explicitly loaded. One could not include it here and demand the user that wants to serialise add this, but how should a user know that internally concatenated_sequences uses a vector?

> I would use if __has_include(<cereal/types/vector.hpp>)

No, this would be bad, because it would make the compiler search the entire include directories every time and also we want to be able to deactivate cereal, even if it is there – for testing, but not only. That's another reason we have a macro.